### ISSUE TITLE
Enable ‘classMap’ mutation

### ISSUE THREAD

**Issue by yinonov** [lit/lit#2107]
## Description
Suggesting `classMap` to abide the monad principles or at least allow its mutation to allow merging or adding more classes.

### Use case
subclassing a component where base class already has a member returning classMap used by the template. It would be better to modify that member  instead of overriding the template which can obviously become a hazard. Say the base class has a `getRenderClasses` it can be easily altered like so
```js
protected override getRenderClasses() {
  return classMap({
    myClass: condition,
	...super.getRenderClasses()
  });
}
```

## Acceptance criteria
Enable `classMap` mutation

**Comment by justinfagnani** [comment]
Directives return a DirectiveResult object which is not mergable in this way. Directive should nearly always be part of the template and not passed into template from an outside function.

The proper way to merge class info object is to spread them be
```ts
protected override getRenderClasses() {
  return {
    myClass: condition,
	...super.getRenderClasses()
  };
}

render() {
  return html`<div class=${classMap(this.getRenderClasses())}`></div>
}
```

In order to do this the base class may have to change to allow subclassing.

**Comment by justinfagnani** [comment]
cc @dfreedm since I think there was a `getRenderClasses` that returned a `classMap` result in MWC.

**Comment by yinonov** [comment]
Thanks for clarifying @justinfagnani. actually aligned our own code to the pattern.
@dfreedm material-components/material-web#2658 can be resolved by material-components/material-web#2659
